src/L1SharedBridge.sol

```diff
@@ -679,175 +442,87 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         // It should be equal to the length of the following:
         // bytes4 function signature + address l1Receiver + uint256 amount + address l2Sender + bytes _additionalData =
         // = 4 + 20 + 32 + 32 + _additionalData.length >= 68 (bytes).
-
-        // So the data is expected to be at least 56 bytes long.
-        require(_l2ToL1message.length >= 56, "ShB wrong msg len"); // wrong message length
-
         (uint32 functionSignature, uint256 offset) = UnsafeBytes.readUint32(_l2ToL1message, 0);
-        if (bytes4(functionSignature) == IMailbox.finalizeEthWithdrawal.selector) {
-            // this message is a base token withdrawal
-            (l1Receiver, offset) = UnsafeBytes.readAddress(_l2ToL1message, offset);
-            (amount, offset) = UnsafeBytes.readUint256(_l2ToL1message, offset);
-            l1Token = BRIDGE_HUB.baseToken(_chainId);
-        } else if (bytes4(functionSignature) == IL1ERC20Bridge.finalizeWithdrawal.selector) {
+        if (bytes4(functionSignature) == IL1ERC20Bridge.finalizeWithdrawal.selector) {
             // We use the IL1ERC20Bridge for backward compatibility with old withdrawals.
 
             // this message is a token withdrawal
 
             // Check that the message length is correct.
             // It should be equal to the length of the function signature + address + address + uint256 = 4 + 20 + 20 + 32 =
             // 76 (bytes).
-            require(_l2ToL1message.length == 76, "ShB wrong msg len 2");
+            require(_l2ToL1message.length == 76, "USDC-ShB wrong msg len 2");
             (l1Receiver, offset) = UnsafeBytes.readAddress(_l2ToL1message, offset);
             (l1Token, offset) = UnsafeBytes.readAddress(_l2ToL1message, offset);
             (amount, offset) = UnsafeBytes.readUint256(_l2ToL1message, offset);
         } else {
-            revert("ShB Incorrect message function selector");
+            revert("USDC-ShB Incorrect message function selector");
         }
     }
 
     /*//////////////////////////////////////////////////////////////
-                            ERA LEGACY FUNCTIONS
+                            UNUSED BUT REQUIRED BY INTERFACE
```

Comment:
You could also change the interface

Reply:
Ohh, good! Didn't want to change the interface. Will do and remove whatever is not needed!